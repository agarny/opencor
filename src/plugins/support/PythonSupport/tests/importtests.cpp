/*******************************************************************************

Copyright (C) The University of Auckland

OpenCOR is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

OpenCOR is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see <https://gnu.org/licenses>.

*******************************************************************************/

//==============================================================================
// Python support import tests
//==============================================================================

#include "../../../../tests/src/testsutils.h"

//==============================================================================

#include "importtests.h"

//==============================================================================

#include <QtTest/QtTest>

//==============================================================================

void ImportTests::tests()
{
    // Some basic tests to make sure that we can import the Python modules that
    // we are supposed to ship

    QStringList output;

    QVERIFY(!OpenCOR::runCli({ "-c", "PythonShell", OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/importtests.py") }, output));
#if defined(Q_OS_WIN) && defined(QT_DEBUG)
    QCOMPARE(output, OpenCOR::fileContents(OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/importtests.out")));
#else
    QCOMPARE(output, QStringList() << QString());
#endif
}

//==============================================================================

QTEST_APPLESS_MAIN(ImportTests)

//==============================================================================
// End of file
//==============================================================================
