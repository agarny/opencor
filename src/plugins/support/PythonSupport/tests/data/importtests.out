The 'ipykernel' module is not present!
The 'IPython' module is not present!
The 'ipywidgets' module is not present!
The 'jupyter_console' module is not present!
The 'jupyterlab' module is not present!
The 'lxml' module is not present!
The 'matplotlib' module is not present!
The 'nbconvert' module is not present!
The 'notebook' module is not present!
The 'qimage2ndarray' module is not present!
The 'qtconsole' module is not present!
The 'scipy' module is not present!
The 'traitlets' module is not present!
